indico/modules/events/surveys/controllers/management/questionnaire.py

```diff
@@ -46,6 +51,85 @@ def _process(self):
+        response.headers['Content-Disposition'] = 'attachment; filename="survey.json"'
+        return response
+
+
+class RHImportSurveyQuestionnaire(RHManageSurveyBase):
+    """Import a questionnaire in JSON format"""
+
+    def _remove_false_values(self, data):
+        # The forms consider a missing key as False (and a False value as True)
+        for key, value in data.items():
+            if value is False:
+                del data[key]
+
+    def _import_data(self, data):
+        if data['version'] != 1:
+            raise ValueError("Unsupported document format")
+
+        # If the survey currently contains only one empty section, remove it.
+        if len(self.survey.items) == 1 and isinstance(self.survey.items[0], SurveySection):
+            db.session.delete(self.survey.items[0])
+
+        for section in data['sections']:
+            self._import_section(section)
+
+    def _import_section(self, data):
+        self._remove_false_values(data)
+        form = SectionForm(formdata=MultiDict(data.items()), csrf_enabled=False)
+        if form.validate():
+            section = add_survey_section(self.survey, form.data)
+            for item in data['content']:
+                self._import_section_item(section, item)
+        else:
+            raise ValueError('Invalid section')
+
+    def _import_section_item(self, section, data):
+        self._remove_false_values(data)
+        if data['type'] == "text":
+            form = TextForm(formdata=MultiDict(data.items()), csrf_enabled=False)
+            if form.validate():
+                add_survey_text(section, form.data)
+            else:
+                raise ValueError('Invalid text item')
+        elif data['type'] == "question":
+            for key, value in data['field_data'].iteritems():
+                if value is not None:
+                    data[key] = value
+            field_cls = get_field_types()[data['field_type']]
+            form_cls = field_cls.config_form
+            form = form_cls(formdata=MultiDict(data.items()), csrf_enabled=False)
+            field_cls.process_imported_data(form)
+            if not form.validate():
+                raise ValueError('Invalid question')
+            add_survey_question(section, field_cls, form.data)
+
+    def _process(self):
+        form = ImportQuestionnaireForm()
+        if form.validate_on_submit():
+            json_text = request.files.getlist('file')[0].file.getvalue()
+            try:
+                data = json.loads(json_text)
+                self._import_data(data)
+            except ValueError as exception:
+                logger.info('%s tried to import an invalid JSON file: %s', session.user, exception.message)
+                flash(_("Invalid file selected."), 'error')
+            else:
+                sections = len(data['sections'])
+                flash(ngettext("{} section has been imported", "{} sections have been imported", sections)
```

Comment:
I'd just show *"The questionnaire has been imported"* - the number of sections is not that important IMO